omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImpl.java

```diff
@@ -632,16 +632,23 @@ private void eliminateCandidateSearchHandlersWithMissingRequiredParameters(Set<S
 		return resourceHandlers;
 	}
 	
-	/* (non-Javadoc)
-	 * @see org.openmrs.module.webservices.rest.web.api.RestService#getSearchHandlers(java.lang.String)
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.api.RestService#getAllSearchHandlers()
+	 * @should return all search handlers if search handlers have been initialized
```

Comment:
And based on the way @should work if they are on a Service the tests should be in ServiceTest, if you really think this behavior is Impl specific then you would need to move the tests to a new RestServiceImplTest class